src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+
+    model_ids = [model.id for model in models.data if model.id.startswith("gpt-")]
+    return judge_model_name in model_ids
+
+
+def run_dk_bench(
+    serve_config: _serve,
+    tls_insecure: bool,
+    tls_client_cert: str,
+    tls_client_key: str,
+    tls_client_passwd: str,
+    model: str,
+    max_workers: str | int | None,
+    gpus: int | None,
+    backend: str | None,
+    enable_serving_output: bool,
+    input_questions: str,
+    system_prompt: str,
+    temperature: float,
+    judge_model_name: str,
+) -> tuple[EvaluationResult, str]:
+    """
+    Runs DK-Bench Evaluation.
+
+    Args:
+        serve_config (_serve):           Name of the judge model to check validity of.
+        tls_insecure (bool):            OpenAI API key.
+        tls_client_cert (str):           OpenAI API key.
+        tls_client_key (str):            OpenAI API key.
+        tls_client_passwd (str):         OpenAI API key.
+        model (str):                     OpenAI API key.
+        max_workers (str | int | None):  OpenAI API key.
+        gpus (int | None):               OpenAI API key.
+        backend (str | None):            OpenAI API key.
+        enable_serving_output (bool):            OpenAI API key.
+        input_questions (str):            OpenAI API key.
+        system_prompt (str):            OpenAI API key.
+        temperature (float):            OpenAI API key.
+        judge_model_name (str):            OpenAI API key.
+    Returns:
+        result (Evaluation):      ragas EvaluationResult to parse for scores
+                                  for summary sheet and questions, references,
+                                  and responses sheet.
+        str:                      Model name of model responses were collected
+                                  from.
+    """
+    if "OPENAI_API_KEY" not in os.environ:
+        raise EnvironmentError(
+            "Environment variable 'OPENAI_API_KEY' must be set to run the Judge model in DK-Bench."
+        )
+    judge_openai_api_key = os.environ.get("OPENAI_API_KEY", "NO_API_KEY")
+
+    if not is_judge_model_name_valid(judge_model_name, judge_openai_api_key):
+        raise ValueError("Judge model name must be a valid OpenAI GPT model")
+
+    # RagasEvaluator.run() expects a Path of a .jsonl file
+    input_questions_path = pathlib.Path(input_questions).resolve()
+    validate_input_questions(input_questions_path)
+
+    try:
+        test_df = pd.read_json(input_questions_path, orient="records", lines=True)
+        if "response" in test_df.columns:
+            logger.info(
+                "Input file %s already contains responses for evaluation. Responses from %s will not be collected for this file.",
+                input_questions_path,
+                model,
+            )
+            get_responses_from_model = False
+        else:
+            get_responses_from_model = True
+
+    except BaseException as exc:
```

Comment:
You want to avoid using `BaseException` generally, but it's fine to use here.

Reply:
This wasn't something I was happy about either but I didn't know what kind of exception would get thrown if `read_json` errored out, so I played it safe with `BaseException`

Reply:
@alimaredia It should be `Exception` as a general rule. `BaseException` should really ought to be hidden.